kamel-image/src/commonMain/kotlin/io/kamel/image/LazyPainterResource.kt

```diff
@@ -52,13 +54,22 @@ public inline fun <I : Any> lazyPainterResource(
             .build()
     }
 
+    val cachedOutput = remember(key) {
+        val output = kamelConfig.mapInput(data)
+        when (data.toString().substringAfterLast(".")) {
+            "svg" -> kamelConfig.svgCache[output]?.let { Resource.Success(it, Memory) }
+            "xml" -> kamelConfig.imageVectorCache[output]?.let { Resource.Success(it, Memory) }
+            else -> kamelConfig.imageBitmapCache[output]?.let { Resource.Success(it, Memory) }
+        }
+    }
+

```

Comment:
In the ImageLoading file could you just make a new public function called something like loadCachedResource that returns the cached resource or null.

Reply:
I'm just being being picky now, but can you name it `cachedResource`? then looks good to me